src/v/cloud_storage/tests/remote_segment_index_test.cc

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "cloud_storage/remote_segment_index.h"
+#include "random/generators.h"
+#include "vlog.h"
+
+#include <seastar/testing/test_case.hh>
+#include <seastar/testing/thread_test_case.hh>
+
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include <stdexcept>
+
+using namespace cloud_storage;
+using namespace cloud_storage::details;
+
+template<class TVal>
+std::vector<TVal> populate_encoder(
+  deltafor_encoder<TVal>& c,
+  uint64_t initial_value,
+  const std::vector<TVal>& deltas) {
+    std::vector<TVal> result;
+    auto p = initial_value;
+    for (TVal delta : deltas) {
+        std::array<TVal, FOR_buffer_depth> buf = {};
+        for (int x = 0; x < FOR_buffer_depth; x++) {
+            result.push_back(p);
+            buf.at(x) = p;
+            p += random_generators::get_int(delta);
+            if (p < buf.at(x)) {
+                throw std::out_of_range("delta can't be represented");
+            }
+        }
+        c.add(buf);
+    }
+    return result;
+}
+
+BOOST_AUTO_TEST_CASE(roundtrip_test_2) {
+    static constexpr int64_t initial_value = 0;
+    deltafor_encoder<int64_t> enc(initial_value);
+    std::vector<int64_t> deltas = {
+      0LL,
+      10LL,
+      100LL,
```

Comment:
are the increasing values of 10x special or should there be some variability in the distance?